adapta/storage/models/aws.py

```diff
@@ -36,7 +36,7 @@ def to_uri(self) -> str:
         if not self.bucket or not self.path:
             raise ValueError("Bucket and path must be defined")
 
-        return f"s3://{self.bucket}/{self.path}"
+        return f"s3a://{self.bucket.rstrip('/')}/{self.path}"
```

Comment:
same for path

Reply:
Added `__post_init__` with a regex expression to check all valid cases for s3 data paths as suggested in [cbf15e9](https://github.com/SneaksAndData/adapta/pull/453/commits/cbf15e9e7b14a169873cfe6abc83a39a2df21bc2)

Since it is a complex regex I suggest looking into the unit tests, I tried to cover most of the corner cases